File: src/CSP-AF/CSPCore/CSPSearchValue.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Views;
using CloudSuitePortal.Services;

using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;

namespace actions.CSP
{
    class CSPSearchValue
    {
        private CSPTable Table { get; set; }
        private CSPInput SearchInput { get; set; }
        private CSPButton SearchButton { get; set; }
        
        public void run(Dictionary<string, object> Params)
        {
            var tableId = (string)Params["Table Id"];
            var columnId = (string)Params["Column Id"];
            var inputId = (string)Params["Search Input Id"];
            var searchValue = (string)Params["Search Value"];
            var buttonId = (string)Params["Search Button Id"];
            var rowIndex = Int32.Parse((string)Params["Row Index"]);
            
            this.Prepare(inputId, buttonId, tableId);
            
            this.SearchInput.SetValue(searchValue);
            this.SearchButton.Click();
            
            var rows = (List<CSPTableRow>)this.Table.Rows;
            var columnElement = rows[rowIndex].WebElementForColumn(columnId);
            
            var wait = new WebDriverWait(CSPBrowser.Application.WebDriver, TimeSpan.FromSeconds(30));
            wait.Until(ExpectedConditions.TextToBePresentInElement(columnElement, searchValue));
            
            this.CleanUp();
        }
        
        public void Prepare(string inputId, string buttonId, string tableId)
        {
            var navigationController = CSPBrowser.Application.WindowController.NavigationController;
            var visibleViewController = navigationController.VisibleViewController;
            
            this.SearchInput = new CSPInput(null, inputId);
            visibleViewController.View.AddSubview(this.SearchInput);
            
            this.SearchButton = new CSPButton(null, buttonId);
            visibleViewController.View.AddSubview(this.SearchButton);
            
            this.Table = new CSPTable(null, tableId);
            visibleViewController.View.AddSubview(this.Table);
        }
        
        public void CleanUp()
        {
            this.SearchInput.RemoveFromSuperview();
            this.SearchButton.RemoveFromSuperview();
            this.Table.RemoveFromSuperview();
        }
    }
}

File: src/CSPAF/CSPCore/CSPSearchValue.java
package CSPAF.CSPCore;

import java.util.*;

import com.infor.cloudsuite.portal.core.views.*;
import com.infor.cloudsuite.portal.core.helpers.*;
import com.infor.cloudsuite.portal.core.services.*;
import com.infor.cloudsuite.portal.core.controllers.*;

import org.openqa.selenium.*;
import org.openqa.selenium.support.ui.*;

class CSPSearchValue
{
    private CSPTable table = null;
    private CSPInput searchInput = null;
    private CSPButton searchButton = null;
        
    public void run(HashMap<String, Object> params)
    {
        String tableId = params.get("Table Id").toString();
        String columnId = params.get("Column Id").toString();
        String inputId = params.get("Search Input Id").toString();
        String searchValue = params.get("Search Value").toString();
        String buttonId = params.get("Search Button Id").toString();
        int rowIndex = Integer.parseInt(params.get("Row Index").toString());
            
        this.prepare(inputId, buttonId, tableId);
        
        this.searchInput.setValue(searchValue);
        this.searchButton.click();
            
        List<CSPTableRow> rows = (List<CSPTableRow>)this.table.getRows();
        WebElement columnElement = rows.get(rowIndex).webElementForColumn(columnId);
            
        WebDriverWait wait = new WebDriverWait(CSPBrowser.getApplication().getWebDriver(), 30);
        wait.until(ExpectedConditions.textToBePresentInElement(columnElement, searchValue));
            
        this.cleanUp();
    }
        
    public void prepare(String inputId, String buttonId, String tableId)
    {
        CSPNavigationController navigationController = CSPBrowser.getApplication().getWindowController().getNavigationController();
        CSPViewController visibleViewController = navigationController.getVisibleViewController();
            
        this.searchInput = new CSPInput(null, inputId);
        visibleViewController.getView().addSubview(this.searchInput);
    
        this.searchButton = new CSPButton(null, buttonId);
        visibleViewController.getView().addSubview(this.searchButton);
            
        this.table = new CSPTable(null, tableId);
        visibleViewController.getView().addSubview(this.table);
    }
        
    public void cleanUp()
    {
        this.searchInput.removeFromSuperview();
        this.searchButton.removeFromSuperview();
        this.table.removeFromSuperview();
    }
}